.github/workflows/ci.yml

```diff
@@ -17,16 +17,89 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: 'npm'
-      - run: npm ci
+      - uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
+      - name: Install dependencies if cache invalid
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: npm ci
+  pr:
+    needs: install
+    strategy:
+      matrix:
+        node: [16.x, 18.x]
+        os: [macos-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      VITE_FR_AM_URL: ${{ secrets.VITE_FR_AM_URL }}
+      VITE_FR_AM_COOKIE_NAME: ${{ secrets.VITE_FR_AM_COOKIE_NAME }}
+      VITE_FR_OAUTH_PUBLIC_CLIENT: ${{ secrets.VITE_FR_OAUTH_PUBLIC_CLIENT }}
+      VITE_FR_REALM_PATH: ${{ secrets.VITE_FR_REALM_PATH }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Restore npm dependencies
+        uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
       - run: npm run build
-        env:
-          VITE_FR_AM_URL: ${{ secrets.VITE_FR_AM_URL }}
-          VITE_FR_AM_COOKIE_NAME: ${{ secrets.VITE_FR_AM_COOKIE_NAME }}
-          VITE_FR_OAUTH_PUBLIC_CLIENT: ${{ secrets.VITE_FR_OAUTH_PUBLIC_CLIENT }}
-          VITE_FR_REALM_PATH: ${{ secrets.VITE_FR_REALM_PATH }}
-      - name: Publish to Chromatic
-        if: ${{ matrix.os == 'ubuntu-latest' && matrix.node < 17 && !github.event.pull_request.draft }}
-        uses: chromaui/action@v1
-        with:
-          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+      - name: Build Storybook
+        run: npm run build:storybook --quiet
+      - run: npx playwright install
+      - name: Serve Storybook and run tests
+        if: ${{ runner.os == 'ubuntu-latest' }} # this one hangs on mac os for some reason
```

Comment:
because i could only get this to run successfully on Ubuntu thats how i have this here.

Reply:
Do we want to create a separate ticket or task to look into fixing this, outside of this PR?

Reply:
I'm okay with this running only on one platform. I don't think there's any value to running it on different OSs when it will use the same browser on each OS: Chrome.